zio-flow/shared/src/main/scala/zio/flow/internal/ZFlowExecutor.scala

```diff
@@ -32,6 +32,28 @@ trait ZFlowExecutor {
    */
   def submit[E: Schema, A: Schema](id: FlowId, flow: ZFlow[Any, E, A]): IO[E, A]
 
+  /**
+   * Submits a flow to be executed and returns a durable promise that will
+   * complete when the flow completes.
+   *
+   * If the executor is already running a flow with the given ID, the existing
+   * flow's durable promise will be returned
+   */
+  def start[E, A](
+    id: FlowId,
+    flow: ZFlow[Any, E, A]
+  ): ZIO[Any, IOException, DurablePromise[Either[Throwable, DynamicValue], DynamicValue]]
+
+  /**
+   * Poll currently running and complete workflows.
+   *
+   * If the workflow with the provided id is completed, it will be returned.
+   */
+  def pollWorkflow[E: Schema, A: Schema](id: FlowId): ZIO[Any, Exception, Option[IO[E, A]]]
+
+  def pollWorkflowDynTyped(id: FlowId): ZIO[Any, Exception, Option[IO[DynamicValue, DynamicValue]]] =
+    pollWorkflow[DynamicValue, DynamicValue](id)
```

Comment:
This will run into `DynamicValue.toTyped[DynamicValue]` in `processResult` which I'm not sure about. If it works and is an identity then it's ok, otherwise you'll have to write another implementation that skips the `processResult` step completely

Reply:
Missed that one, indeed: will add an alternative implementation.

Reply:
(Though, it worked in at least one case, as illustrated by the unit test in PersistentExecutorSpec at line 109(